paymentsheet/src/test/java/com/stripe/android/paymentsheet/model/PaymentSelectionTest.kt

```diff
@@ -17,92 +17,57 @@ class PaymentSelectionTest {
-            val link = PaymentSelection.Link
-            val result = link.mandateText(
-                merchantName = "Merchant",
-                isSaveForFutureUseSelected = isSaveForFutureUseSelected,
-                isSetupFlow = false,
-            )
-            assertThat(result).isNull()
-        }
-
-    @Test
-    fun `Doesn't display a mandate for Google Pay`() =
-        runAllConfigurations { isSaveForFutureUseSelected ->
-            val googlePay = PaymentSelection.GooglePay
-            val result = googlePay.mandateText(
-                merchantName = "Merchant",
-                isSaveForFutureUseSelected = isSaveForFutureUseSelected,
-                isSetupFlow = false,
-            )
-            assertThat(result).isNull()
-        }
+    fun `Doesn't display a mandate for Link`() {
+        val link = PaymentSelection.Link
+        val result = link.mandateText(
+            merchantName = "Merchant",
+            isSetupFlow = false,
+        )
+        assertThat(result).isNull()
+    }
 
     @Test
-    fun `Doesn't display a mandate for new US bank accounts`() =
-        runAllConfigurations { isSaveForFutureUseSelected ->
-            // We actually do show a mandate, but it's set independently from the PaymentSelection.
-            val newPaymentSelection = PaymentSelection.New.USBankAccount(
-                labelResource = "Test",
-                iconResource = 0,
-                paymentMethodCreateParams = mock(),
-                customerRequestedSave = mock(),
-                input = PaymentSelection.New.USBankAccount.Input(
-                    name = "",
-                    email = null,
-                    phone = null,
-                    address = null,
-                    saveForFutureUse = false,
-                ),
-                instantDebits = null,
-                screenState = USBankAccountFormScreenState.SavedAccount(
-                    financialConnectionsSessionId = "session_1234",
-                    intentId = "intent_1234",
-                    bankName = "Stripe Bank",
-                    last4 = "6789",
-                    primaryButtonText = "Continue".resolvableString,
-                    mandateText = null,
-                ),
-            )
-
-            val result = newPaymentSelection.mandateText(
-                merchantName = "Merchant",
-                isSaveForFutureUseSelected = isSaveForFutureUseSelected,
-                isSetupFlow = false,
-            )
-
-            assertThat(result).isNull()
-        }
+    fun `Doesn't display a mandate for Google Pay`() {
+        val googlePay = PaymentSelection.GooglePay
+        val result = googlePay.mandateText(
+            merchantName = "Merchant",
+            isSetupFlow = false,
+        )
+        assertThat(result).isNull()
+    }
 
     @Test
-    fun `Displays the correct mandate for a saved US bank account when saving for future use`() {
```

Comment:
This case was never possible in practice, so I removed the code that triggered it, and removed the test that validated it.